internal/perms/perms.go

```diff
@@ -17,35 +17,21 @@ package perms
 
 import (
 	"fmt"
-	"os/user"
-	"strconv"
+	"os"
 )
 
 // MyUserAndGroup returns the UID and GID of this process.
 func MyUserAndGroup() (uid uint32, gid uint32, err error) {
-	// Ask for the current user.
-	user, err := user.Current()
-	if err != nil {
-		err = fmt.Errorf("Fetching current user: %w", err)
-		return
-	}
-
-	// Parse UID.
-	uid64, err := strconv.ParseUint(user.Uid, 10, 32)
-	if err != nil {
-		err = fmt.Errorf("Parsing UID (%s): %w", user.Uid, err)
-		return
-	}
+	signed_uid := os.Getuid()
+	signed_gid := os.Getgid()
 
-	// Parse GID.
-	gid64, err := strconv.ParseUint(user.Gid, 10, 32)
-	if err != nil {
-		err = fmt.Errorf("Parsing GID (%s): %w", user.Gid, err)
+	if signed_gid == -1 || signed_uid == -1 {
```

Comment:
the doc says it returns -1 in windows https://pkg.go.dev/os#Getgid. Did we confirm for unix ?

Reply:
It works in RHEL 8, haven't tested in CentOs if that's what you meant by Unix. If something else, do let me know.

```
gargnitin@gargnitin:~/work/cloud/storage/client/gcsfuse/tasks/202307-golang1.20.5/solutions/nocgo/proj1$ CGO_ENABLED=0 go build && (ldd ./proj1 || true)
	not a dynamic executable
gargnitin@gargnitin:~/work/cloud/storage/client/gcsfuse/tasks/202307-golang1.20.5/solutions/nocgo/proj1$ docker run -v $PWD:/src -w /src -u 431:433 -it registry.access.redhat.com/ubi8/ubi:8.1 sh -c "(ldd ./proj1 || true) && ./proj1"
	not a dynamic executable
UID, GID from os.user.Current():  <nil> user: Current requires cgo or $USER set in environment
UID, GID from os.Get(u/g)id():  431 433
```

Reply:
@sethiay  ^

Reply:
Just to clarify, I want to confirm if os.Getuid() returns -1 when it fails to get uid ?  I am confirming because the documentation doesn't talk about -1 case https://pkg.go.dev/os#Getgid

Reply:
> when it fails to get uid ?

What do you mean by that? Do you have a way to create that scenario? If yes, we might be able to test it ourselves.

> I am confirming because the documentation doesn't talk about -1 case

It mentions windows, but that's beside your question.

Reply:
Agreed to capture the open question in a comment here with @sethiay .